sovrin_client/agent/endpoint.py

```diff
@@ -11,27 +11,40 @@
 logger = getlogger()
 
 
-class Endpoint(SimpleStack):
+class Endpoint:
     def __init__(self, port: int, msgHandler: Callable,
                  name: str=None, basedirpath: str=None):
+
         if name and basedirpath:
             ha = getHaFromLocalEstate(name, basedirpath)
             if ha and ha[1] != port:
                 port = ha[1]
 
+        self.name = name
+        self.stack = None
+
+        self._port = port
+        self._msgHandler = msgHandler
+        self._basedirpath = basedirpath
+
+    def start(self):
         stackParams = {
```

Comment:
You can simply check whether the stack was already created stopped, then just start it. No need to create it again.